editor/editor_file_system.cpp

```diff
@@ -816,55 +898,59 @@ void EditorFileSystem::_scan_new_dir(EditorFileSystemDirectory *p_dir, Ref<DirAc
 	dirs.sort_custom<FileNoCaseComparator>();
 	files.sort_custom<FileNoCaseComparator>();
 
-	int total = dirs.size() + files.size();
-	int idx = 0;
+	int nb_files_total_scan = 0;
 
-	for (List<String>::Element *E = dirs.front(); E; E = E->next(), idx++) {
+	for (List<String>::Element *E = dirs.front(); E; E = E->next()) {
 		if (da->change_dir(E->get()) == OK) {
 			String d = da->get_current_dir();
 
 			if (d == cd || !d.begins_with(cd)) {
 				da->change_dir(cd); //avoid recursion
 			} else {
-				EditorFileSystemDirectory *efd = memnew(EditorFileSystemDirectory);
-
-				efd->parent = p_dir;
-				efd->name = E->get();
+				ScannedDirectory *sd = memnew(ScannedDirectory);
+				sd->name = E->get();
 
-				_scan_new_dir(efd, da, p_progress.get_sub(idx, total));
+				nb_files_total_scan += _scan_new_dir(sd, da);
 
-				int idx2 = 0;
-				for (int i = 0; i < p_dir->subdirs.size(); i++) {
-					if (efd->name.filenocasecmp_to(p_dir->subdirs[i]->name) < 0) {
-						break;
-					}
-					idx2++;
-				}
-				if (idx2 == p_dir->subdirs.size()) {
-					p_dir->subdirs.push_back(efd);
-				} else {
-					p_dir->subdirs.insert(idx2, efd);
-				}
+				p_dir->subdirs.push_back(sd);
 
 				da->change_dir("..");
 			}
 		} else {
 			ERR_PRINT("Cannot go into subdir '" + E->get() + "'.");
 		}
+	}
+
+	p_dir->files = files;
+	nb_files_total_scan += files.size();
+
+	return nb_files_total_scan;
+}
+
+void EditorFileSystem::_process_file_system(const ScannedDirectory *p_scan_dir, EditorFileSystemDirectory *p_dir, const ScanProgress &p_progress) {
+	String dir_full_path = p_dir->get_path();
```

Comment:
`get_path()` computes the path by traversing its parent folders, and also does so in a very suboptimal way.
I wonder of it's worth to cache the full path String at the time of creation. Or at least `get_path()` could be changed to use StringBuilder.

Reply:
I added a full_path member to ScannedDirectory and it is calculated in `_scan_new_dir`. I was able to remove the path has a parameter of `_first_scan_process_scripts`.
Done!